### PR TITLE
rust: Simplify and update dependencies

### PR DESCRIPTION
**Pull request by cdecker**
Some of the dependencies were getting a bit long in the teeth, so let's update them here.

I also added a simplification, in that we rely on the `bitcoin` crate to pull in the `bitcoin_hashes` and `secp256k1` crates, so at least they shouldn't get out of sync, and we can bump just that one dependency.

This also bumps the cln-* crate versions, so we can publish them as soon as this PR gets merged/

Changelog-None No user-visible changes in the API in this PR (prior PRs waiting to be published do, but they have their own entries)

### CODE REVIEW

File: cln-grpc/src/pb.rs [hunk 1]
```diff
@@ -31,7 +31,7 @@ impl From<JOutpoint> for Outpoint {
 impl From<Outpoint> for JOutpoint {
     fn from(a: Outpoint) -> Self {
         JOutpoint {
-            txid: bitcoin_hashes::sha256::Hash::from_slice(&a.txid).unwrap(),
+            txid: bitcoin::hashes::sha256::Hash::from_slice(&a.txid).unwrap(),
```
Comment (vincenzopalazzo) [inline cln-grpc/src/pb.rs:34]:
```suggestion
            txid: Hash::from_slice(&a.txid).unwrap(),
```

**Review by vincenzopalazzo — COMMENTED** [review]
I just go curious from the PR and give a read :)

File: cln-grpc/src/pb.rs (hunk 1 above)
Comment (cdecker) [reply cln-grpc/src/pb.rs:34]: I'm afraid not: `bitcoin::hashes::Hash` is a trait, whereas `bitcoin::hashes::sha256::Hash` is a struct implementing that trait.

Comment (vincenzopalazzo) [reply cln-grpc/src/pb.rs:34]: Oh man, this is confusing, yeah I got the point now!

**Comment by RCasatta** [comment]
I think this supersede #5753

**Comment by cdecker** [comment]
ACK https://github.com/ElementsProject/lightning/pull/5820/commits/bdac1e73b2fc5fad26a52c19d98a0ec1fe0a89e7

**Pull request merged**